controllers/storageclient_controller.go

```diff
@@ -177,110 +155,95 @@ func (s *StorageClientReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	return result, err
 }
 
-func (s *StorageClientReconciler) reconcilePhases(instance *v1alpha1.StorageClient) (ctrl.Result, error) {
+func (r *StorageClientReconciler) reconcilePhases() (ctrl.Result, error) {
 	storageClientListOption := []client.ListOption{
-		client.MatchingFields{storageProviderEndpointIndexName: instance.Spec.StorageProviderEndpoint},
+		// TODO: this isn't fool proof, need to revisit as a storageclient can be created to same provider but via different endpoint incase of nodeport
+		client.MatchingFields{storageProviderEndpointIndexName: r.storageClient.Spec.StorageProviderEndpoint},
 	}
 
 	storageClientList := &v1alpha1.StorageClientList{}
-	if err := s.Client.List(s.ctx, storageClientList, storageClientListOption...); err != nil {
-		s.Log.Error(err, "unable to list storage clients")
+	if err := r.Client.List(r.ctx, storageClientList, storageClientListOption...); err != nil {
+		r.Log.Error(err, "unable to list storage clients")
 		return ctrl.Result{}, err
 	}
 
 	if len(storageClientList.Items) > 1 {
-		s.Log.Info("one StorageClient is allowed per namespace but found more than one. Rejecting new request.")
-		instance.Status.Phase = v1alpha1.StorageClientFailed
+		r.Log.Info("one StorageClient is allowed with a unique storageproviderendpoint but found more than one. Rejecting new request.")
+		r.storageClient.Status.Phase = v1alpha1.StorageClientFailed
```

Comment:
Why are you trying to protect against 2 clients on the same cluster connecting to the same provider twice? I can see use cases for such configurations and I don't; see any reason to disallow it. 

The only limitation should be that we should not allow two client CR having the same name. But that is already covered by k8s itself.

Reply:
I didn't impose the restriction, I added a comment to surface the obvious code surrounding it. Need to check why it was imposed in the first place.

Reply:
It was imposed for 2 clients on the same namespace as we wanted a namespace to govern a single client. That is not the case anymore.

Reply:
It is, but why does that matter?

Reply:
It matters because off boarding is per storageconsumer, isn't it?

Reply:
raised #142